snowflake/data_access.go

```diff
@@ -233,35 +226,57 @@ func (s *AccessSyncer) SyncAccessAsCodeToTarget(ctx context.Context, access map[
 		return err
 	}
 
-	rolesToRemove := make([]string, 0)
+	rolesToRemove := make(map[string]*importer.AccessProvider, 0)
 
 	for role, toKeep := range existingRoles {
 		if !toKeep {
-			rolesToRemove = append(rolesToRemove, role)
+			rolesToRemove[role] = access[role]
 		}
 	}
 
-	err = s.removeRolesToRemove(rolesToRemove, repo)
+	err = s.removeRolesToRemove(rolesToRemove, repo, nil)
 	if err != nil {
 		return err
 	}
 
-	err = s.generateAccessControls(ctx, access, existingRoles, repo, configMap)
+	err = s.generateAccessControls(ctx, access, existingRoles, repo, configMap, nil)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (s *AccessSyncer) removeRolesToRemove(rolesToRemove []string, repo dataAccessRepository) error {
+func (s *AccessSyncer) removeRolesToRemove(rolesToRemove map[string]*importer.AccessProvider, repo dataAccessRepository, feedbackHandler wrappers.AccessProviderFeedbackHandler) error {
 	if len(rolesToRemove) > 0 {
-		logger.Info(fmt.Sprintf("Removing old Raito roles in Snowflake: %s", rolesToRemove))
+		logger.Info(fmt.Sprintf("Removing %d old Raito roles in Snowflake", len(rolesToRemove)))
+
+		for roleToRemove, roleAp := range rolesToRemove {
+			externalId := roleToRemove
+
+			var fi *importer.AccessProviderSyncFeedback
+			if feedbackHandler != nil && roleAp != nil {
```

Comment:
1. Maybe we should always create the `AccessProviderSyncFeedback` object and only at the end check if we need to provide it (or create a dummy handler)
2. Is it possible that roleAp is `nil`?